### PR TITLE
Add i18n date

### PR DESCRIPTION
**Pull request by ofa-**
- [x] add i18n'd date+time formatter (as per Prefs `locale` and `12/24h` time settings)
- [x] use i18n'd date+time format
- [x] `dateFormatFull` now usused => remove
- [x] `timeFormat{12,24}hrs` now unused => remove

### CODE REVIEW

**Comment by jeevatkm** [comment]
@ofa- Thank you for the PR. I will go through and get back to you.

**Comment by ofa-** [comment]
@jeevatkm will rebase this PR on top of your master - and there are typos in commit messages

**Comment by ofa-** [comment]
@jeevatkm I rebased this PR on top of current master.  Anything preventing you from merging that I could help with ?

File: content/rwh.js [hunk 1]
```diff
@@ -981,6 +976,31 @@ var ReplyWithHeader = {
     }
   },
 
+  DateFmt: function(d_str) {
```
Comment (jeevatkm) [inline content/rwh.js]: @ofa- Please use `camelCase` for variables. It seems func `DateFmt` accepts `date` object not a string.  May be something like `function (date) {`.

File: content/rwh.js [hunk 2]
```diff
@@ -981,6 +976,31 @@ var ReplyWithHeader = {
...
+    var options = {
+        weekday: "short",
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+    }
+    var time_opt = {
```
Comment (jeevatkm) [inline content/rwh.js]: @ofa- Please use `camelCase` for variables. May be something like `timeOpts`.

Question; is it beneficial to declare variable `options` & `timeOpts` outside the function?

File: content/rwh.js [hunk 3]
```diff
@@ -981,6 +976,31 @@ var ReplyWithHeader = {
...
+        hour: "numeric",
+        minute: "numeric",
+        hour12: (this.Prefs.timeFormat == 0),
+    };
+    var locale = this.Prefs.headerLocale;
+
+    var date = new Date(d_str);
```
Comment (jeevatkm) [inline content/rwh.js]: Per previous comment, change it to `date`.

File: content/rwh.js [hunk 4]
```diff
@@ -981,6 +976,31 @@ var ReplyWithHeader = {
...
+
+    var dateStr = date.toLocaleDateString(locale, options);
+    var timeStr = date.toLocaleTimeString(locale, time_opt);
```
Comment (jeevatkm) [inline content/rwh.js]: Per previous comment, change it to `timeOpts`

**Comment by ofa-** [comment]
I fixed-up the PR with the requested changes, original branch is available at `add_i18n_date.orig`
Below is the diff:

```
ofa:~/.thunderbird/replyWithHeader$ git diff add_i18n_date.orig 
diff --git a/content/rwh.js b/content/rwh.js
index 240680e..3482eec 100644
--- a/content/rwh.js
+++ b/content/rwh.js
@@ -976,24 +976,24 @@ var ReplyWithHeader = {
     }
   },
 
-  DateFmt: function(d_str) {
+  DateFmt: function(date) {
     var options = {
         weekday: "short",
         year: "numeric",
         month: "short",
         day: "numeric",
     }
-    var time_opt = {
+    var timeOpts = {
         hour: "numeric",
         minute: "numeric",
         hour12: (this.Prefs.timeFormat == 0),
     };
     var locale = this.Prefs.headerLocale;
 
-    var date = new Date(d_str);
+    date = new Date(date);
 
     var dateStr = date.toLocaleDateString(locale, options);
-    var timeStr = date.toLocaleTimeString(locale, time_opt);
+    var timeStr = date.toLocaleTimeString(locale, timeOpts);
 
     return {
         date: dateStr,
```

**Comment by ofa-** [comment]
ah, just saw you merged the PR 45 and added another commit on top
=> rebased this PR on top of your new master.

**Comment by jeevatkm** [comment]
@ofa- Thank you, merging it :)

**Comment by jeevatkm** [comment]
@ofa- May I know your full name? I would like to use it for release notes.

**Comment by ofa-** [comment]
part of #12

**Pull request merged**